### DISCUSSION TITLE
Simulating Non ROADM solution

### DISCUSSION THREAD

**Discussion by SaiBhyri** [Telecominfraproject/oopt-gnpy#478] [Q&A]
Hi Team,

Could you please help on how we simulate a non-roadm solution i.e. TxP-> Passive Optics -> Amp -> Fiber? What are the changes required in Equipment JSON?

Thanks,
Sai

**Comment by EstherLerouzic** [comment]
Dear @SaiBhyri 
the JSON template may allow you to create such a line
I think that this example might work for you:
https://github.com/Telecominfraproject/oopt-gnpy/blob/master/gnpy/example-data/edfa_example_network.json
it describes transceiver -> fiber -> amplifier -> transceiver
to describe a passive optics you might use the Fused type which represents a concentrated loss. I only models loss.

Hope it helps,

regards

Esther

**Comment by SaiBhyri** [comment]
Thanks Esther for your response.
Sorry, probably i havent described my query properly.

Network Scenario is as described below: 
Site 1 (Transceiver -> Passive Optics -> Bst Amplifier) <---  Span1 ---> Site 2 (Pre Amp -> Passive Optics -> Transceiver)

How many "Sites" should i create in my input excel and what should be their "Type"?
Are any specific rules to be followed during Span creation and Eqpt forcing in excel for this scenario?
It will be very helpful if you could help create a sample input excel for this scenario.

Thanks,
Sai

**Comment by SaiBhyri** [comment]
Hi Esther,

Could you please help on my above query?

Thanks,
Sai

**Comment by EstherLerouzic** [comment]
Dear Sai,
I do not recommand to use the excel converter to create this type of topology. The excel input was developped for ROADM type topologies with some constraints and automatic fill-in that you don't want in your type of topologies. For example, with the excel template:
- you can not create a topology without at least two ROADMs,
- each non ROADM elements in Nodes sheet is created with east and west instance, so that the topology has the two direction, 
- each line in the Links sheet adds a fiber between Node A and Node Z in east and west direction,
- ...

You may want to use the excel to create all those elements and then filter out only those elements that fit to your target topology, removing  all unwanted connections, and elements. And this can be only post process, with potential risk of errors if you do it manually.

I tried to do that (manually), and I found the process quite painful even for this really simple topology.

Here is the excel I used, the json topology file automatically created on this topology that I used as a template, and the final topology. 
`gnpy-transmission-example example_for_sai.xls --save-network-before-autodesign example_for_sai.json`

[example_for_Sai.json](https://github.com/Telecominfraproject/oopt-gnpy/files/13611606/example_for_Sai.json)
[example_for_Sai.xls](https://github.com/Telecominfraproject/oopt-gnpy/files/13611608/example_for_Sai.xls)
[example_for_Sai_corrected.json](https://github.com/Telecominfraproject/oopt-gnpy/files/13611609/example_for_Sai_corrected.json)
 

So again my recommandation is to create those topologies directly with JSON format, and not use excel format.

**Comment by SaiBhyri** [comment]
Thanks Esther for your kind response.
JSON format seems very painful for huge networks.
With excel, its easy to copy paste data from customer files. 
Is there any future plan to ease this process?

Does the online documentation contain all the constraints with excel and which need to be take care of?

Thanks,
Sai
